<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>HERMES 3.0 sex-stratified analysis • heRmes</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="HERMES 3.0 sex-stratified analysis">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">heRmes</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9000</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/hermes_3_codes.html">HERMES 3.0 sex-stratified analysis</a></li>
    <li><a class="dropdown-item" href="../articles/nih_cardiomyopathy.html">Cardiomyopathy phenotyping NIH</a></li>
    <li><a class="dropdown-item" href="../articles/ukbb_phenotyping.html">UKBB phenotyping example</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>HERMES 3.0 sex-stratified analysis</h1>
                        <h4 data-toc-skip class="author">Dr. Clara
Jiang</h4>
            
      

      <div class="d-none name"><code>hermes_3_codes.Rmd</code></div>
    </div>

    
    
<style>
body {
text-align: justify}
</style>
<div class="section level2">
<h2 id="aim">Aim<a class="anchor" aria-label="anchor" href="#aim"></a>
</h2>
<p>To perform a multi-ancestry sex-stratified genome-wide association
study (GWAS) meta-analysis of HF, ischaemic HF and non-ischaemic HF.</p>
</div>
<div class="section level2">
<h2 id="rationale">Rationale<a class="anchor" aria-label="anchor" href="#rationale"></a>
</h2>
<p>Heart failure (HF) has shown substantial sex differences in its
clinical presentations and risk profiles, but much remains unclear about
HF in women.<span class="citation"><sup><a href="#ref-groenewegenEpidemiologyHeartFailure2020">1</a>–<a href="#ref-kenchaiahHeartFailureWomen2015">3</a></sup></span>
Sex-stratified genome-wide association studies (GWAS) have provided
mechanistic insights into diseases with known phenotypic sex
differences.<span class="citation"><sup><a href="#ref-cardiogramplusc4dconsortiumLargescaleAssociationAnalysis2013">4</a>,<a href="#ref-boerDecipheringOsteoarthritisGenetics2021">5</a></sup></span>
This project aims to identify sex-specific HF risk loci, by performing a
sex-stratified GWAS of all-cause HF and subtypes stratified by ischaemic
aetiology, including an analysis of the X chromosome (ChrX). This
research, particularly the analysis of non-ischaemic HF, is envisioned
to build new knowledge on the sex-specific genetic basis of HF and
identify non-traditional risk factors.</p>
<p>HF amongst individuals with congenital heart disease represents a
distinct entity, largely due to structural malformations during
embryonic cardiogenesis. Whilst most cases remain unexplained, genetic
association studies have highlighted variants in important patterning
genes as potential causal factors.<span class="citation"><sup><a href="#ref-lahmCongenitalHeartDisease2021">6</a></sup></span> In this
study, we are interested in why hearts that seemingly developed normally
then go on to fail. Given the often dramatic structural abnormalities
seen in congenital heart disease and likely distinct pathophysiology
(embryonic patterning), we have excluded this subgroup from our HF
definition.</p>
<p>Chronic respiratory disease, predominantly COPD, is a common
comorbidity amongst patients with diagnosed HF.<span class="citation"><sup><a href="#ref-ehteshami-afsharClinicalCharacteristicsOutcomes2021">7</a></sup></span>
Whilst outcomes are poorer in this HF subgroup, standard HF therapeutics
offer similar prognostic benefits suggesting shared underlying
pathophysiological mechanisms.<span class="citation"><sup><a href="#ref-ehteshami-afsharClinicalCharacteristicsOutcomes2021">7</a></sup></span>
Diagnosis of either, or both, conditions can be challenging due to
overlapping clinical signs and symptoms. However, chronic respiratory
disease and HF share multiple upstream risk factors - smoking being the
most obvious - therefore conditioning on chronic respiratory disease
status has the potential to induce bias in the data. Therefore, we have
not excluded chronic respiratory disease patients from our HF
definition, as long as they also have a definite, non-ambiguous, HF
diagnostic code.</p>
</div>
<div class="section level2">
<h2 id="study-phenotypes">Study phenotypes<a class="anchor" aria-label="anchor" href="#study-phenotypes"></a>
</h2>
<table class="table">
<colgroup>
<col width="14%">
<col width="20%">
<col width="64%">
</colgroup>
<thead><tr class="header">
<th>Phenotype</th>
<th>Name</th>
<th>Brief Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>Phenotype 1</td>
<td>All HF</td>
<td>Clinical syndrome of HF, any cause or manifestation</td>
</tr>
<tr class="even">
<td>Phenotype 2</td>
<td>Ischaemic HF</td>
<td>HF comorbid with antecedent CAD</td>
</tr>
<tr class="odd">
<td>Phenotype 3</td>
<td>Non-ischaemic HF</td>
<td>HF without CAD</td>
</tr>
</tbody>
</table>
</div>
<div class="section level2">
<h2 id="methods">Methods<a class="anchor" aria-label="anchor" href="#methods"></a>
</h2>
<p><em>Subjects:</em> Individuals of European, South Asian, East Asian,
African, Hispanic, and Indigenous American ancestries who are free of
congenital heart disease, and with or without prevalent HF.</p>
<p><em>Genotypes:</em> Autosomal and ChrX SNPs (preferably imputed
against HRC, TOPMED, or population-specific haplotypes).</p>
<p><em>Phenotypes:</em> Three phenotypes as described above (minimum of
500 cases and 500 controls in each sex for each ancestry-specific GWAS
analysis).</p>
</div>
<div class="section level2">
<h2 id="return-of-results">Return of results<a class="anchor" aria-label="anchor" href="#return-of-results"></a>
</h2>
<p>The Cardiovascular Knowledge Portal team are developing a private
data repository for HERMES3 with functionality to validate file formats
and run QC on cohort-level GWAS summary data. If our initial pilot study
is successful, we will switch to this platform. In the meantime, please,
return results for your study to the analysis team at UQ via the UQRDM.
To get access to UQRDM, please provide your analyst email address to
Clara Jiang at <a href="mailto:j.jiang@uq.edu.au" class="email">j.jiang@uq.edu.au</a>.</p>
</div>
<div class="section level2">
<h2 id="phenotype-definition">Phenotype definition<a class="anchor" aria-label="anchor" href="#phenotype-definition"></a>
</h2>
<p>To define phenotypes in your study, please follow a two-step approach
described below:</p>
<div class="section level3">
<h3 id="step-1">Step 1<a class="anchor" aria-label="anchor" href="#step-1"></a>
</h3>
<p>For each participant, define the status (TRUE / FALSE) for the
following phenotype classifiers:</p>
<table class="table">
<colgroup>
<col width="21%">
<col width="12%">
<col width="65%">
</colgroup>
<thead><tr class="header">
<th>Classifier</th>
<th>Classifier ID</th>
<th>Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>Congenital heart disease</td>
<td>CongHD</td>
<td>Any congenital malformation of the heart or great vessels defined by
a diagnosis, or relevant surgical or percutaneous procedure</td>
</tr>
<tr class="even">
<td>Heart failure</td>
<td>HF</td>
<td>Diagnosis of heart failure of any cause</td>
</tr>
<tr class="odd">
<td>Coronary artery disease</td>
<td>CAD</td>
<td>Diagnosis of myocardial infarction, history of coronary thrombolysis
procedure and surgical or percutaneous procedure for coronary artery
revascularization</td>
</tr>
</tbody>
</table>
<p>Where available, manually adjudicated clinical definitions made by
trained physicians are preferred.</p>
<p>For de novo adjudication using electronic health records, please
define these classifiers as the presence of at least 1 relevant
diagnosis/procedure code listed in Appendix Ia-c. Where appropriate,
please also include other code systems (e.g. READ and CPT) by mapping
the recommended codes.</p>
<p>If self-reported data are available, individuals who lack a HF or CAD
diagnosis/procedure code, but self-report to have had these conditions,
should be excluded from the study.</p>
<p>Please contact the central analyst team for cohort-specific
adjudication not mentioned here.</p>
</div>
<div class="section level3">
<h3 id="step-2">Step 2<a class="anchor" aria-label="anchor" href="#step-2"></a>
</h3>
<p>For each participant, classify the status (case/control/exclude) for
target GWAS phenotypes by combining classifier status from Step 1 using
the logic described below.</p>
<p><em>Table 1. GWAS Phenotype classifiers for all-cause HF, ischaemic
HF, and non-ischaemic HF (phenotypes 1-3)</em></p>
<table class="table">
<thead><tr class="header">
<th>CongHD</th>
<th>HF</th>
<th>CAD</th>
<th>Category</th>
<th>Phenotype</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>TRUE</td>
<td>ANY</td>
<td>ANY</td>
<td>Exclude</td>
<td></td>
</tr>
<tr class="even">
<td>FALSE</td>
<td>TRUE</td>
<td>TRUE</td>
<td>Ischaemic HF case</td>
<td>1 &amp; 2</td>
</tr>
<tr class="odd">
<td>FALSE</td>
<td>TRUE</td>
<td>FALSE</td>
<td>Non-ischaemic HF case</td>
<td>1 &amp; 3</td>
</tr>
<tr class="even">
<td>FALSE</td>
<td>FALSE</td>
<td>TRUE</td>
<td>Exclude</td>
<td></td>
</tr>
<tr class="odd">
<td>FALSE</td>
<td>FALSE</td>
<td>FALSE</td>
<td>CONTROL</td>
<td></td>
</tr>
</tbody>
</table>
<p><em>Figure 1. Phenotyping algorithm</em></p>
<p><img src="figures/phenotyping_algorithm.png" align="centre" width="80%"></p>
</div>
</div>
<div class="section level2">
<h2 id="obtaining-concept-codes">Obtaining concept codes<a class="anchor" aria-label="anchor" href="#obtaining-concept-codes"></a>
</h2>
<p>Load the <code>heRmes</code> R package.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://nicksunderland.github.io/heRmes/">heRmes</a></span><span class="op">)</span></span></code></pre></div>
<p>Obtain the codes with the <code>get_codes</code> function. The codes
are available within the package structure at:
<code>inst/extdata/ukhdr_phenotypes</code> and from the <a href="https://phenotypes.healthdatagateway.org" class="external-link">UKHDR Phenotype
Library</a> website.</p>
<div class="section level3">
<h3 id="ukhdr-phenoptype-codes">UKHDR phenoptype codes<a class="anchor" aria-label="anchor" href="#ukhdr-phenoptype-codes"></a>
</h3>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hermes3_pheno_ids</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>congenital_heart_disease <span class="op">=</span> <span class="st">"PH1645"</span>, </span>
<span>                          heart_failure            <span class="op">=</span> <span class="st">"PH1645"</span>, </span>
<span>                          coronary_artery_disease  <span class="op">=</span> <span class="st">"PH1645"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="congenital-heart-disease">Congenital heart disease<a class="anchor" aria-label="anchor" href="#congenital-heart-disease"></a>
</h3>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">chd</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_codes.html">get_codes</a></span><span class="op">(</span>pheno_id <span class="op">=</span> <span class="va">hermes3_pheno_ids</span><span class="op">[[</span><span class="st">"congenital_heart_disease"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"phenotype_id"</span>, <span class="st">"phenotype_name"</span>, <span class="st">"coding_system.name"</span>, <span class="st">"code"</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="va">chd</span><span class="op">[</span><span class="op">]</span></span>
<span><span class="co">#&gt;    phenotype_id phenotype_name coding_system.name   code</span></span>
<span><span class="co">#&gt;          &lt;char&gt;         &lt;char&gt;             &lt;char&gt; &lt;char&gt;</span></span>
<span><span class="co">#&gt; 1:       PH1645  Heart failure         ICD9 codes  40201</span></span>
<span><span class="co">#&gt; 2:       PH1645  Heart failure         ICD9 codes  42832</span></span>
<span><span class="co">#&gt; 3:       PH1645  Heart failure         ICD9 codes  42821</span></span>
<span><span class="co">#&gt; 4:       PH1645  Heart failure         ICD9 codes  42823</span></span>
<span><span class="co">#&gt; 5:       PH1645  Heart failure         ICD9 codes  42820</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="heart-failure">Heart failure<a class="anchor" aria-label="anchor" href="#heart-failure"></a>
</h3>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hf</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_codes.html">get_codes</a></span><span class="op">(</span>pheno_id <span class="op">=</span> <span class="va">hermes3_pheno_ids</span><span class="op">[[</span><span class="st">"heart_failure"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"phenotype_id"</span>, <span class="st">"phenotype_name"</span>, <span class="st">"coding_system.name"</span>, <span class="st">"code"</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="va">hf</span><span class="op">[</span><span class="op">]</span></span>
<span><span class="co">#&gt;    phenotype_id phenotype_name coding_system.name   code</span></span>
<span><span class="co">#&gt;          &lt;char&gt;         &lt;char&gt;             &lt;char&gt; &lt;char&gt;</span></span>
<span><span class="co">#&gt; 1:       PH1645  Heart failure         ICD9 codes  40201</span></span>
<span><span class="co">#&gt; 2:       PH1645  Heart failure         ICD9 codes  42832</span></span>
<span><span class="co">#&gt; 3:       PH1645  Heart failure         ICD9 codes  42821</span></span>
<span><span class="co">#&gt; 4:       PH1645  Heart failure         ICD9 codes  42823</span></span>
<span><span class="co">#&gt; 5:       PH1645  Heart failure         ICD9 codes  42820</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="coronary-artery-disease">Coronary artery disease<a class="anchor" aria-label="anchor" href="#coronary-artery-disease"></a>
</h3>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cad</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_codes.html">get_codes</a></span><span class="op">(</span>pheno_id <span class="op">=</span> <span class="va">hermes3_pheno_ids</span><span class="op">[[</span><span class="st">"coronary_artery_disease"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"phenotype_id"</span>, <span class="st">"phenotype_name"</span>, <span class="st">"coding_system.name"</span>, <span class="st">"code"</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="va">cad</span><span class="op">[</span><span class="op">]</span></span>
<span><span class="co">#&gt;    phenotype_id phenotype_name coding_system.name   code</span></span>
<span><span class="co">#&gt;          &lt;char&gt;         &lt;char&gt;             &lt;char&gt; &lt;char&gt;</span></span>
<span><span class="co">#&gt; 1:       PH1645  Heart failure         ICD9 codes  40201</span></span>
<span><span class="co">#&gt; 2:       PH1645  Heart failure         ICD9 codes  42832</span></span>
<span><span class="co">#&gt; 3:       PH1645  Heart failure         ICD9 codes  42821</span></span>
<span><span class="co">#&gt; 4:       PH1645  Heart failure         ICD9 codes  42823</span></span>
<span><span class="co">#&gt; 5:       PH1645  Heart failure         ICD9 codes  42820</span></span></code></pre></div>
</div>
</div>
<div class="section level2 unnumbered">
<h2 class="unnumbered" id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body">
<div id="ref-groenewegenEpidemiologyHeartFailure2020" class="csl-entry">
<div class="csl-left-margin">1. </div>
<div class="csl-right-inline">Groenewegen A, Rutten FH, Mosterd A, Hoes AW.
Epidemiology of heart failure. <em>European Journal of Heart
Failure</em>. 2020;22(8):1342-1356. doi:<a href="https://doi.org/10.1002/ejhf.1858" class="external-link">10.1002/ejhf.1858</a>
</div>
</div>
<div id="ref-heidenreich2022AHAACC2022" class="csl-entry">
<div class="csl-left-margin">2. </div>
<div class="csl-right-inline">Heidenreich PA, Bozkurt B, Aguilar D, Allen LA,
Byun JJ, Colvin MM, Deswal A, Drazner MH, Dunlay SM, Evers LR, Fang JC,
Fedson SE, Fonarow GC, Hayek SS, Hernandez AF, Khazanie P, Kittleson MM,
Lee CS, Link MS, Milano CA, Nnacheta LC, Sandhu AT, Stevenson LW,
Vardeny O, Vest AR, Yancy CW. 2022
<span>AHA</span>/<span>ACC</span>/<span>HFSA Guideline</span> for the
<span>Management</span> of <span>Heart Failure</span>: <span>A
Report</span> of the <span>American College</span> of
<span>Cardiology</span>/<span>American Heart Association Joint
Committee</span> on <span>Clinical Practice Guidelines</span>.
<em>Circulation</em>. 2022;145(18):e895-e1032. doi:<a href="https://doi.org/10.1161/CIR.0000000000001063" class="external-link">10.1161/CIR.0000000000001063</a>
</div>
</div>
<div id="ref-kenchaiahHeartFailureWomen2015" class="csl-entry">
<div class="csl-left-margin">3. </div>
<div class="csl-right-inline">Kenchaiah S, Vasan RS. Heart
<span>Failure</span> in <span>Women–Insights</span> from the
<span>Framingham Heart Study</span>. <em>Cardiovascular Drugs and
Therapy</em>. 2015;29(4):377-390. doi:<a href="https://doi.org/10.1007/s10557-015-6599-0" class="external-link">10.1007/s10557-015-6599-0</a>
</div>
</div>
<div id="ref-cardiogramplusc4dconsortiumLargescaleAssociationAnalysis2013" class="csl-entry">
<div class="csl-left-margin">4. </div>
<div class="csl-right-inline">CARDIoGRAMplusC4D Consortium, Deloukas P,
Kanoni S, Willenborg C, Farrall M, Assimes TL, Thompson JR, Ingelsson E,
Saleheen D, Erdmann J, Goldstein BA, Stirrups K, König IR, Cazier JB,
Johansson A, Hall AS, Lee JY, Willer CJ, Chambers JC, Esko T, Folkersen
L, Goel A, Grundberg E, Havulinna AS, Ho WK, Hopewell JC, Eriksson N,
Kleber ME, Kristiansson K, Lundmark P, Lyytikäinen LP, Rafelt S, Shungin
D, Strawbridge RJ, Thorleifsson G, Tikkanen E, Van Zuydam N, Voight BF,
Waite LL, Zhang W, Ziegler A, Absher D, Altshuler D, Balmforth AJ,
Barroso I, Braund PS, Burgdorf C, Claudi-Boehm S, Cox D, Dimitriou M, Do
R, DIAGRAM Consortium, CARDIOGENICS Consortium, Doney ASF, El Mokhtari
N, Eriksson P, Fischer K, Fontanillas P, Franco-Cereceda A, Gigante B,
Groop L, Gustafsson S, Hager J, Hallmans G, Han BG, Hunt SE, Kang HM,
Illig T, Kessler T, Knowles JW, Kolovou G, Kuusisto J, Langenberg C,
Langford C, Leander K, Lokki ML, Lundmark A, McCarthy MI, Meisinger C,
Melander O, Mihailov E, Maouche S, Morris AD, Müller-Nurasyid M, MuTHER
Consortium, Nikus K, Peden JF, Rayner NW, Rasheed A, Rosinger S, Rubin
D, Rumpf MP, Schäfer A, Sivananthan M, Song C, Stewart AFR, Tan ST,
Thorgeirsson G, et al. Large-scale association analysis identifies new
risk loci for coronary artery disease. <em>Nature Genetics</em>.
2013;45(1):25-33. doi:<a href="https://doi.org/10.1038/ng.2480" class="external-link">10.1038/ng.2480</a>
</div>
</div>
<div id="ref-boerDecipheringOsteoarthritisGenetics2021" class="csl-entry">
<div class="csl-left-margin">5. </div>
<div class="csl-right-inline">Boer
CG, Hatzikotoulas K, Southam L, Stefánsdóttir L, Zhang Y, Coutinho de
Almeida R, Wu TT, Zheng J, Hartley A, Teder-Laving M, Skogholt AH, Terao
C, Zengini E, Alexiadis G, Barysenka A, Bjornsdottir G, Gabrielsen ME,
Gilly A, Ingvarsson T, Johnsen MB, Jonsson H, Kloppenburg M, Luetge A,
Lund SH, Mägi R, Mangino M, Nelissen RRGHH, Shivakumar M, Steinberg J,
Takuwa H, Thomas LF, Tuerlings M, arcOGEN Consortium, HUNT All-In Pain,
ARGO Consortium, Regeneron Genetics Center, Babis GC, Cheung JPY, Kang
JH, Kraft P, Lietman SA, Samartzis D, Slagboom PE, Stefansson K,
Thorsteinsdottir U, Tobias JH, Uitterlinden AG, Winsvold B, Zwart JA,
Davey Smith G, Sham PC, Thorleifsson G, Gaunt TR, Morris AP, Valdes AM,
Tsezou A, Cheah KSE, Ikegawa S, Hveem K, Esko T, Wilkinson JM,
Meulenbelt I, Lee MTM, van Meurs JBJ, Styrkársdóttir U, Zeggini E.
Deciphering osteoarthritis genetics across 826,690 individuals from 9
populations. <em>Cell</em>. 2021;184(18):4784-4818.e17. doi:<a href="https://doi.org/10.1016/j.cell.2021.07.038" class="external-link">10.1016/j.cell.2021.07.038</a>
</div>
</div>
<div id="ref-lahmCongenitalHeartDisease2021" class="csl-entry">
<div class="csl-left-margin">6. </div>
<div class="csl-right-inline">Lahm
H, Jia M, Dreßen M, Wirth F, Puluca N, Gilsbach R, Keavney BD, Cleuziou
J, Beck N, Bondareva O, Dzilic E, Burri M, König KC, Ziegelmüller JA,
Abou-Ajram C, Neb I, Zhang Z, Doppler SA, Mastantuono E, Lichtner P,
Eckstein G, Hörer J, Ewert P, Priest JR, Hein L, Lange R, Meitinger T,
Cordell HJ, Müller-Myhsok B, Krane M. Congenital heart disease risk loci
identified by genome-wide association study in <span>European</span>
patients. <em>The Journal of Clinical Investigation</em>.
2021;131(2):e141837, 141837. doi:<a href="https://doi.org/10.1172/JCI141837" class="external-link">10.1172/JCI141837</a>
</div>
</div>
<div id="ref-ehteshami-afsharClinicalCharacteristicsOutcomes2021" class="csl-entry">
<div class="csl-left-margin">7. </div>
<div class="csl-right-inline">Ehteshami-Afshar S, Mooney L, Dewan P, Desai
AS, Lang NN, Lefkowitz MP, Petrie MC, Rizkala AR, Rouleau JL, Solomon
SD, Swedberg K, Shi VC, Zile MR, Packer M, McMurray JJV, Jhund PS,
Hawkins NM. Clinical <span>Characteristics</span> and
<span>Outcomes</span> of <span>Patients With Heart Failure With Reduced
Ejection Fraction</span> and <span>Chronic Obstructive Pulmonary
Disease</span>: <span>Insights From PARADIGM-HF</span>. <em>Journal of
the American Heart Association</em>. 2021;10(4):e019238. doi:<a href="https://doi.org/10.1161/JAHA.120.019238" class="external-link">10.1161/JAHA.120.019238</a>
</div>
</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Nicholas Sunderland.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

    </footer>
</div>





  </body>
</html>
